src/spatialexperiment/SpatialExperiment.py

```diff
@@ -0,0 +1,814 @@
+        output += f"row_names({0 if self._row_names is None else len(self._row_names)}): {' ' if self._row_names is None else ut.print_truncated_list(self._row_names)}\n"
+
+        output += f"column_data columns({len(self._cols.column_names)}): {ut.print_truncated_list(self._cols.column_names)}\n"
+        output += f"column_names({0 if self._column_names is None else len(self._column_names)}): {' ' if self._column_names is None else ut.print_truncated_list(self._column_names)}\n"
+
+        output += f"main_experiment_name: {' ' if self._main_experiment_name is None else self._main_experiment_name}\n"
+        output += f"reduced_dims({len(self.reduced_dim_names)}): {ut.print_truncated_list(self.reduced_dim_names)}\n"
+        output += f"alternative_experiments({len(self.alternative_experiment_names)}): {ut.print_truncated_list(self.alternative_experiment_names)}\n"
+        output += f"row_pairs({len(self.row_pair_names)}): {ut.print_truncated_list(self.row_pair_names)}\n"
+        output += f"column_pairs({len(self.column_pair_names)}): {ut.print_truncated_list(self.column_pair_names)}\n"
+
+        output += f"metadata({str(len(self.metadata))}): {ut.print_truncated_list(list(self.metadata.keys()), sep=' ', include_brackets=False, transform=lambda y: y)}\n"
+
+        return output
+
+    ##############################
+    #####>> spatial_coords <<#####
+    ##############################
+
+    def get_spatial_coordinates(self) -> biocframe.BiocFrame:
+        """Access spatial coordinates.
+
+        Returns:
+            A ``BiocFrame`` containing columns of spatial coordinates.
+        """
+        return self._spatial_coords
+
+    def get_spatial_coords(self) -> biocframe.BiocFrame:
+        """Alias for :py:meth:`~get_spatial_coordinates`."""
+        return self.get_spatial_coordinates()
+
+    def set_spatial_coordinates(
+        self, _spatial_coords: Optional[biocframe.BiocFrame], in_place: bool = False
+    ) -> "SpatialExperiment":
+        """Set new spatial coordinates.
+
+        Args:
+            _spatial_coords:
+                :py:class:`~biocframe.BiocFrame.BiocFrame` containing columns of spatial coordinates. Must be the same length as `column_data`. Typical column names might include:
+
+                    - **['x', 'y']**: For simple 2D coordinates.
+                    - **['pxl_col_in_fullres', 'pxl_row_in_fullres']**: For pixel-based coordinates in full-resolution images.
+
+                Spatial coordinates are coerced to a
+                :py:class:`~biocframe.BiocFrame.BiocFrame`.
+
+            in_place:
+                Whether to modify the ``SpatialExperiment`` in place. Defaults to False.
+
+        Returns:
+            A modified ``SpatialExperiment`` object, either as a copy of the original or as a reference to the (in-place-modified) original.
+        """
+        spatial_coords = _sanitize_frame(_spatial_coords)
```

Comment:
Something to check is if `_spatial_coords` can be None. With some of the other props, e.g. column data can set be set to None. Does the R class allow this behavior and if so, does it create an empty biocframe with n rows?

Reply:
seems like thats the case by calling `_sanitize_frame` bcoz is creates an empty dataframe, but the docstring should specify that one can set it to None to remove the coordinate information.

Also if so, then the [_sanitize_frame](https://github.com/BiocPy/SummarizedExperiment/blob/master/src/summarizedexperiment/_frameutils.py#L10) function expects a number of rows parameter.